src/sentry/utils/eventuser.py

```diff
@@ -35,11 +47,80 @@ def from_event(event: Event):
             username=event.data.get("user", {}).get("username") if event else None,
             name=event.data.get("user", {}).get("name") if event else None,
             ip_address=event.data.get("user", {}).get("ip_address") if event else None,
+            user_id=event.data.get("user", {}).get("id") if event else None,
         )
 
     def get_display_name(self):
         return self.name or self.email or self.username
 
+    @classmethod
+    def attr_from_keyword(self, keyword):
+        return KEYWORD_MAP.get_key(keyword)
+
+    @classmethod
+    def for_projects(self, projects: List[Project], keyword_filters: Mapping[str, Any]):
+        oldest_project = min(projects, key=lambda item: item.date_added)
+
+        where_conditions = [
+            Condition(Column("project_id"), Op.IN, [p.id for p in projects]),
+            Condition(Column("timestamp"), Op.LT, datetime.now()),
+            Condition(Column("timestamp"), Op.GTE, oldest_project.date_added),
+        ]
+
+        for keyword, value in keyword_filters.items():
+            snuba_column = KEYWORD_MAP.get_key(keyword)
+            if isinstance(snuba_column, tuple):
+                for column in snuba_column:
+                    where_conditions.append(Condition(Column(column), Op.EQ, value))
+            else:
+                where_conditions.append(Condition(Column(snuba_column), Op.EQ, value))
+
+        query = Query(
+            match=Entity(EntityKey.Events.value),
+            select=[
+                Column("project_id"),
+                Column("group_id"),
+                Column("ip_address_v6"),
+                Column("ip_address_v4"),
+                Column("event_id"),
+                Column("user_id"),
+                Column("user"),
+                Column("user_name"),
+                Column("user_email"),
+            ],
+            where=where_conditions,
+        )
+
+        request = Request(
+            dataset=Dataset.Events.value,
+            app_id=REFERRER,
+            query=query,
+            tenant_ids={"referrer": REFERRER, "organization_id": projects[0].organization.id},
+        )
+        data_results = raw_snql_query(request, referrer=REFERRER)["data"]
+        results = [EventUser.from_snuba(result) for result in data_results]
```

Comment:
This will return a lot of rows that you don't really need. You really just want the most recent event imo, since you're throwing the rest of the data away in `get_user_tag`

That will hopefully make the query a little faster too, since it can return as soon as it finds a row.